Upgrade to srcML 1.0 - Groundwork update 3

cnewman opened the pull request:
This pull request addresses the following:
-Use of SRCMLBINDIR has been ceased and removed.
-Errors are now at least partially handled (went with system exception for now). Typically, the behavior is to throw up into the API and expect the API to handle exceptions from the wrapper.
-Multiple files and archives are fully handled in the base srcML query functions. Can give it multiple files/archives and can receive them in return.
-Implementation of srcML archive and unit modifying functions is complete (but untested)
-Issue #56 should be handled. The original function was split into two functions; one for unit and one for archive.
-Name of the cpp side of the project was changed to LibSrcMLWrapper
-Name of C# side (the API) was changed to LibSrcMLRunner

vinayaugustine commented on ABB.SrcML/LibSrcMLRunner.cs:
```diff
@@ -1,10 +1,12 @@
 using System.Runtime.InteropServices;
 namespace ABB.SrcML {
-    public class SrcMLCppAPI {
+    public class LibSrcMLRunner {
```
Consider applying the [`CLSCompliant(false)`](https://msdn.microsoft.com/en-us/library/12a7a7h3%28v=vs.120%29.aspx#CLSAttribute) attribute to `LibSrcMLRunner` -- this will clear up a bunch of the

vinayaugustine commented:
The [Src2SrcMLPreview](https://github.com/cnewman/SrcML.NET/blob/cb4e842a66d07673fa3b68bc767bfe9052a2bb69/Tools/ABB.SrcML.Tools.Src2SrcMLPreview/Src2SrcMLPreviewWindow.xaml.cs#L64) project is broken. It refers to `SrcMLHelper.GetSrcMLDefaultDirectory()`. Looks like it should be updated to `SrcMLHelper.GetSrcMLRootDirectory()`

cnewman commented:
@vinayaugustine Do you want the CLSCompliant change and the Src2SrcMLPreview change added to the pull request?

vinayaugustine commented:
Yeah, that’d be great.

From: cnewman [mailto:notifications@github.com]
Sent: Wednesday, September 16, 2015 10:58 AM
To: abb-iss/SrcML.NET
Cc: Vinay Augustine
Subject: Re: [SrcML.NET] Upgrade to srcML 1.0 - Groundwork update 3 (#58)


@vinayaugustine<https://github.com/vinayaugustine> Do you want the CLSCompliant change and the Src2SrcMLPreview change added to the pull request?

—
Reply to this email directly or view it on GitHub<https://github.com/abb-iss/SrcML.NET/pull/58#issuecomment-140767223>.

cnewman commented:
@vinayaugustine I added the changes.

The non-compliance thing is actually addressed in the next pull request since the .dll is 32 bit. I ended up having to move from ulong to uint32. Regardless, will leave the compliance attribute in since I don't know that it'll be guaranteed moving into the future.

Updated the function call in srcMLpreview too.
